src/query/plan/rule_based_planner.hpp

```diff
@@ -569,62 +583,55 @@ class RuleBasedPlanner {
   std::unique_ptr<LogicalOperator> GenFilters(std::unique_ptr<LogicalOperator> last_op,
                                               const std::unordered_set<Symbol> &bound_symbols, Filters &filters,
                                               AstStorage &storage, const SymbolTable &symbol_table) {
-    std::unique_ptr<LogicalOperator> pattern_filter =
-        ExtractPatternFilters(filters, storage, symbol_table, bound_symbols);
+    auto pattern_filters = ExtractPatternFilters(filters, symbol_table, storage, bound_symbols);
```

Comment:
since you are using pattern filters only when filter expr exists, does it make sense to move it inside `if{}`

Reply:
I understand now that this doesn't work, as `impl::ExtractFilters(bound_symbols, filters, storage);` will delete your filter used.

Reply:
Just confirm this, and I will resolve comment

Reply:
Confirmed